TrackingTools/TransientTrack/interface/CandidatePtrTransientTrack.h

```diff
@@ -72,9 +76,18 @@ namespace reco {
 
     TrajectoryStateClosestToBeamLine stateAtBeamLine() const;
 
+    void setTime(double t, double dt) {
+      hasTime = true; timeExt_ = t; dtErrorExt_ = dt;
+    }
```

Comment:
here and elewhere: with the setTime method available, do we need all the alternative constructors?